index.bs

```diff
@@ -2097,15 +2099,15 @@ the value is the IDL {{undefined}} value.
              :  {{DOMString}}
              :  an [=enumeration=] type
              :: The value of the <emu-t class="regex"><a href="#prod-string">string</a></emu-t> token
-                is the sequence of 16 bit unsigned integer code units
-                (hereafter referred to just as <dfn id="dfn-code-unit">code units</dfn>)
+                is the sequence of 16 bit unsigned integer [=code units=]
                 corresponding to the UTF-16 encoding of |S|.
              :  {{ByteString}}
              :: The value of the <emu-t class="regex"><a href="#prod-string">string</a></emu-t> token
                 is the sequence of 8 bit unsigned integer code units
                 corresponding to the UTF-8 encoding of |S|.
```

Comment:
If we want to continue to treat these as byte sequences (I could see an argument for isomorphic string, though that would be a bigger change with downstream impact on Fetch and such), we should use assert that |S| doesn't contain any code points higher than U+00FF and use isomorphic encode. UTF-8 is plain wrong.

Reply:
I don't see a definition for "isomorphic string"? I added the assertion and isomorphic encoding. The assertion is redundant with one inside https://infra.spec.whatwg.org/#isomorphic-encode, but I think it's worth having one here too, so people know what they can't write in default arguments.